### DISCUSSION TITLE
How to do partitioned writes on exisiting partitioned_db?

### DISCUSSION THREAD

**Discussion by dkalra88** [duckdb/duckdb#11749] [Q&A] [answered]
I have a Hive partitioned database partitioned by month (partitioned_db). For month of April the data in Hive partitioned database is available till April 15th. The remaining data from April 15th to 30th is in  part2.duckdb. If I use following command from part2.duckdb :

COPY table TO 'partitioned_db' (FORMAT PARQUET, PARTITION_BY (MONTH_NAME), OVERWRITE_OR_IGNORE, FILENAME_PATTERN "table_{i}"); 

then the existing data from April 1st to April 15th is over-written. How can I add the new data to existing partitioned db without over-writing the old data?

**Comment by intelmatt** [comment, marked as answer]
It looks to me like perhaps DuckDB cannot do incremental updates of Hive-partitioned data natively. But, one option is to merge the two Hives outside of DuckDB. Let's say you have hive1/ which is your large dataset, and then you have a new table called new_data. You can write new_data to a new hive as such:

COPY new_data TO 'hive_tmp' (FORMAT PARQUET, PARTITION_BY (xxxx), FILENAME_PATTERN "data_{uuid}");

The data_ file should have a unique name because of uuid in the FILENAME_PATTERN. Then you can use rsync to merge hive_tmp/ into hive1/:

rsync -av hive_tmp/* hive1/
